fe/fe-core/src/main/java/com/starrocks/server/OlapTableFactory.java

```diff
@@ -266,9 +267,15 @@ public Table createTable(LocalMetastore metastore, Database db, CreateTableStmt
                     PropertyAnalyzer.analyzeBooleanProp(properties, PropertyAnalyzer.PROPERTIES_ENABLE_PERSISTENT_INDEX,
                             false);
             if (enablePersistentIndex && table.isCloudNativeTable()) {
-                throw new DdlException("Cannot create cloud native table with persistent index yet");
-            }
+                // Judge there are whether compute nodes or not.
+                // Cannot create cloud native table with persistent_index = true with ComputeNode
+                if (GlobalStateMgr.getCurrentSystemInfo().getIdComputeNode().size() != 0) {
+                    throw new DdlException("Cannot create cloud native table with persistent_index = true with ComputeNode");
```

Comment:
Cannot create cloud native table with local persistent index with ComputeNode ?

Reply:
BE must set storage_path, so only compute node  will be check.  Storage_path info can't be got through heartbeart so far ,  i will add.